terraform/modules/critical/rdsupgrade.tf

```diff
@@ -0,0 +1,84 @@
+locals {
+  rds_archivematica_url = "mysql://${var.rds_username}:${var.rds_password}@${local.rds_host}:${local.rds_port}"
+
+  rds_host = aws_rds_cluster.archivematica.endpoint
+  rds_port = aws_rds_cluster.archivematica.port
+
+  prod_cluster_identifier    = "archivematica"
+  staging_cluster_identifier = "archivematica-${var.namespace}"
+
+  cluster_identifier = var.namespace == "prod" ? local.prod_cluster_identifier : local.staging_cluster_identifier
+
+  prod_database_name    = "archivematica"
+  staging_database_name = "archivematica_${var.namespace}"
+
+  database_name = var.namespace == "prod" ? local.prod_database_name : local.staging_database_name
+
+  snapshot_identifier = "cluster-snapshot-2025-02" # Current snapshot identifier
+}
+
+resource "aws_db_subnet_group" "archivematica" {
+  subnet_ids = var.network_private_subnets
+}
+
+resource "aws_rds_cluster" "archivematica" {
```

Comment:
I think you might have an issue with these resource path e.g. `aws_rds_cluster. archivematica` as they have the same name as existing resources.

Reply:
You may want to use the rds module in `modules.rds` for this.